demo/trackLoader.js

```diff
@@ -138,193 +138,112 @@ async function parseTracks(bytesArray, shaderMaterial, FlatbufferModule, animati
-          sumY += bbox.y();
-          sumZ += bbox.z();
-        }
-        centroidLocation = new THREE.Vector3( sumX/8.0, sumY/8.0, sumZ/8.0 );
-        if (firstCentroid == undefined) {
-          firstCentroid = centroidLocation;
-        }
-        // debugger; // delta
-        delta = centroidLocation.clone().sub(firstCentroid);
+      const state = track.states(ii);
 
-        let p0 = new THREE.Vector3(state.bbox(0).x(), state.bbox(0).y(), state.bbox(0).z()); // Front Left Bottom Point (near front left tire on vehicle e.g.)
-        let p1 = new THREE.Vector3(state.bbox(1).x(), state.bbox(1).y(), state.bbox(1).z()); // Front Right Bottom Point (near front right tire on vehicle e.g.)
-        let p2 = new THREE.Vector3(state.bbox(2).x(), state.bbox(2).y(), state.bbox(2).z()); // Back Right Bottom Point (near back right tire on vehicle e.g.)
-        let p3 = new THREE.Vector3(state.bbox(3).x(), state.bbox(3).y(), state.bbox(3).z());
-        let p4 = new THREE.Vector3(state.bbox(4).x(), state.bbox(4).y(), state.bbox(4).z()); // Front Left Top Point (above front left tire at height of roof of a vehicle e.g.)
-        let p5 = new THREE.Vector3(state.bbox(5).x(), state.bbox(5).y(), state.bbox(5).z());
-        let p6 = new THREE.Vector3(state.bbox(6).x(), state.bbox(6).y(), state.bbox(6).z());
-        let p7 = new THREE.Vector3(state.bbox(7).x(), state.bbox(7).y(), state.bbox(7).z());
+      // Initializations:
+      let centroidLocation;
+      let sumX = 0, sumY=0, sumZ=0;
+      for (let jj=0; jj<8;jj++) {
+        const bbox = state.bbox(jj);
 
+        sumX += bbox.x();
+        sumY += bbox.y();
+        sumZ += bbox.z();
+      }
+      centroidLocation = new THREE.Vector3( sumX/8.0, sumY/8.0, sumZ/8.0 );
 
+      if (firstCentroid == undefined) {
+        firstCentroid = centroidLocation;
+      }
 
-        let length = p2.distanceTo(p1);
-        let width = p1.distanceTo(p0);
-        let height = 2; // TODO Remove once bbox vertices are fixed
-        // let height = p4.distanceTo(p0);
+      const delta = centroidLocation.clone().sub(firstCentroid);
 
-        // let length = state.bbox(2) - state.bbox(1); // Length is the distance from the front to rear bumper of a car
-        // let width = state.bbox(1) - state.bbox(0); // Width is the distance from the driver-side door to the passenger-side door of a car
-        // let height = state.bbox(4) - state.bbox(0); // Height is the distance from the bottom of the tire to the roof of a car
+      let p0 = new THREE.Vector3(state.bbox(0).x(), state.bbox(0).y(), state.bbox(0).z()); // Front Left Bottom Point (near front left tire on vehicle e.g.)
+      let p1 = new THREE.Vector3(state.bbox(1).x(), state.bbox(1).y(), state.bbox(1).z()); // Front Right Bottom Point (near front right tire on vehicle e.g.)
+      let p2 = new THREE.Vector3(state.bbox(2).x(), state.bbox(2).y(), state.bbox(2).z()); // Back Right Bottom Point (near back right tire on vehicle e.g.)
```

Comment:
Probably `let` => `const`. Here and elsewhere.

Reply:
Sounds good, I fixed a bunch but I guess I missed some haha. I'll ctrl+f for `let`